### PR TITLE
Fix/kal select

### PR DESCRIPTION
**Pull request by Bellico**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts [hunk 1]
```diff
@@ -184,11 +184,11 @@ export class KalSelectComponent
   /**
    * Select an option by his value
    */
-  select(value: any): void {
+  select(value: any, notify = false): void {
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts]: could you explain where you need this feature ?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts [hunk 2]
```diff
@@ -261,15 +261,19 @@ export class KalSelectComponent
    * Event emitted when an option is selected
    * Set the option as active
    * @param option KalOptionComponent
+   * @param withNotify boolean
    */
-  private optionSelected(option: KalOptionComponent) {
+  private optionSelected(option: KalOptionComponent, withNotify: boolean = true) {
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts]: consistency : you should use same parameters name for same action

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-option/kal-option.component.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div class="kal-option-selection" (click)="emitSelectionEvent()"
      [class.active]="active"
-     [class.highligh]="isHighlighted"
+     [class.highlight]="isHighlighted"
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-option/kal-option.component.html]: kal-option-highlighted

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-option/kal-option.component.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
      [class.disabled]="disabled">
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-option/kal-option.component.html]: kal-option-disabled

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-option/kal-option.component.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <div class="kal-option-selection" (click)="emitSelectionEvent()"
      [class.active]="active"
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-option/kal-option.component.html]: kal-option-active

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts (hunk 2 above)
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts]: you can just set a value, typescript will type the parameter automatically

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts (hunk 1 above)
Comment (Bellico) [reply projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts]: select method is call on writevalue method, at this moment we should not notify the control

**Review by MichaelGrosio — DISMISSED** [review]

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts [hunk 6]
```diff
@@ -261,15 +261,19 @@ export class KalSelectComponent
...
     } else {
       this.optionSelectedOnSimpleMode(option);
     }
 
-    super.notifyUpdate(this.selectedValue);
+    if (withNotify) {
+      super.notifyUpdate(this.selectedValue);
+    }
+
     this.valueChange.emit(this.selectedValue);
```
Comment (ydomenjoud) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-select/kal-select.component.ts]: should be in withNotify block

**Review by ydomenjoud — APPROVED** [review]

**Pull request merged**